### PR TITLE
fix: bugs with chapter visability due to quest links

### PR DESCRIPTION
**Pull request by MichaelHillcox**
Also removed Lat heads.

Looks like this might not be an issue on 1.20 as your changes have rewritten most of these areas.

### CODE REVIEW

File: common/src/main/java/dev/ftb/mods/ftbquests/gui/quests/ViewQuestPanel.java [hunk 1]
```diff
@@ -437,7 +437,7 @@ private void addDescriptionText(boolean canEdit, Component subtitle) {
 	}
 
 	private void addButtonBar(boolean canEdit) {
-		// button bar has page navigation buttons for multi-page text, and the Add button in edit mode
+		// button bar hasVisibleQuest page navigation buttons for multi-page text, and the Add button in edit mode
```
Comment (desht) [inline common/src/main/java/dev/ftb/mods/ftbquests/gui/quests/ViewQuestPanel.java:437]: search and replace? :smile:

Comment (MichaelHillcox) [reply common/src/main/java/dev/ftb/mods/ftbquests/gui/quests/ViewQuestPanel.java:437]: Damn it! I new I would miss one. That's intellij's stupid "You renamed something common so let me go refactor all the comments"... I'll fix it in a bit

File: common/src/main/java/dev/ftb/mods/ftbquests/gui/quests/ChapterPanel.java [hunk 2]
```diff
@@ -297,7 +297,7 @@ public void draw(PoseStack matrixStack, Theme theme, int x, int y, int w, int h)
 
 			GuiHelper.setupDrawing();
 
-			if (chapter.quests.isEmpty()) {
+			if (chapter.hasVisibleQuestLazy()) {
```
Comment (desht) [inline common/src/main/java/dev/ftb/mods/ftbquests/gui/quests/ChapterPanel.java:297]: Negated check here.  This should be `if (!chapter.hasVisibleQuestLazy())`

**Review by desht — APPROVED** [review]
All good!

**Pull request merged**